src/main/java/frc/team2412/robot/subsystem/ClimbSubsystem.java

```diff
@@ -1,12 +1,107 @@
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
-
 import edu.wpi.first.wpilibj.DoubleSolenoid;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
+import static frc.team2412.robot.subsystem.ClimbSubsystem.ClimbConstants.*;
+
 public class ClimbSubsystem extends SubsystemBase {
-    public ClimbSubsystem(WPI_TalonFX climbFixed1, WPI_TalonFX climbFixed2, WPI_TalonFX climbAngled1, WPI_TalonFX climbAngled2, DoubleSolenoid climbAngle) {
 
+    public static class ClimbConstants {
+        public static final double MAX_SPEED = 1;
+        public static final double TEST_SPEED = 0.5;
+        public static final double STOP_SPEED = 0;
+        public static final double MAX_ENCODER_TICKS = 1000;
+        public static final double MIN_ENCODER_TICKS = 0;
+
+        public static final SupplyCurrentLimitConfiguration MOTOR_CURRENT_LIMIT = new SupplyCurrentLimitConfiguration(true, 40, 40, 500);
+        
+        enum HookArmState { ANGLED, UPRIGHT }
+        enum ClimbSubsystemState { ENABLED, DISABLED }
+    }
+
+    private final WPI_TalonFX climbFixedMotor;
+    private final WPI_TalonFX climbDynamicMotor;
+
+    private final DoubleSolenoid solenoid;
+
+    private static ClimbSubsystemState state = ClimbSubsystemState.DISABLED;
+
+    public ClimbSubsystem(WPI_TalonFX climbFixedMotor, WPI_TalonFX climbDynamicMotor, DoubleSolenoid climbAngle) {
+        setName("ClimbSubsystem");
+        this.climbFixedMotor = climbFixedMotor;
+        this.climbDynamicMotor = climbDynamicMotor;
+        solenoid = climbAngle;
+        climbFixedMotor.configSupplyCurrentLimit(MOTOR_CURRENT_LIMIT);
+        climbDynamicMotor.configSupplyCurrentLimit(MOTOR_CURRENT_LIMIT);
+        climbFixedMotor.configForwardSoftLimitThreshold(MAX_ENCODER_TICKS);
+        climbFixedMotor.configReverseSoftLimitThreshold(MIN_ENCODER_TICKS);
+        climbFixedMotor.configForwardSoftLimitEnable(true, 0);
+        climbFixedMotor.configReverseSoftLimitEnable(true, 0);
+        climbDynamicMotor.configForwardSoftLimitThreshold(MAX_ENCODER_TICKS);
+        climbDynamicMotor.configReverseSoftLimitThreshold(MIN_ENCODER_TICKS);
+        climbDynamicMotor.configReverseSoftLimitEnable(true, 0);
+        climbDynamicMotor.configForwardSoftLimitEnable(true, 0);
+    }
+
+    @Override
+    public void periodic() { 
+        double positionDynamic = climbDynamicMotor.getSelectedSensorPosition();
+        if (!(positionDynamic >= MAX_ENCODER_TICKS || positionDynamic <= MIN_ENCODER_TICKS)) {
+            stopAngledArm();
+        }
+        double positionFixed = climbFixedMotor.getSelectedSensorPosition();
+        if (!(positionFixed >= MAX_ENCODER_TICKS || positionFixed <= MIN_ENCODER_TICKS)) {
+            stopFixedArm();
+        }
+    }
+
+    public void setEnabled() {
+        state = ClimbSubsystemState.ENABLED;
+    }
+
+    public void setDisabled() {
+        state = ClimbSubsystemState.DISABLED;
+    }
```

Comment:
What are these for?  They don't seem to reflect or change any physical state of the robot.

Reply:
It's a state intended to make sure that the drivers don't accidentally cause the robot to climb, so the idea is that they would first press a button to get into climb mode, during endgame and then they would press the climb button or whatever we decide on doing for the climb automation